File: Basics LumApps/nodeJS/oauth2.js

const axios = require('axios');
const oauth = require('axios-oauth-client');
const request = require('request-promise');
const security = require('./security');

// Prepare your request
// In this case, I have placed my confidential information inside an external file called "security"
const getClientCredentials = oauth.client(axios.create(), {
  url: 'https://www.googleapis.com/oauth2/v4/token',
  grant_type: 'refresh_token',
  client_id: security.CLIENT_ID,
  client_secret: security.CLIENT_SECRET,
  refresh_token: security.REFRESH_TOKEN,
  scope: 'https://www.googleapis.com/auth/userinfo.email'
});
 
//Define your async method to request your access_token
async function getToken(){
const auth = await getClientCredentials(); 
const response = await request({
  uri: 'https://lumsites.appspot.com/_ah/api/lumsites/v1/user/get',
  qs: {
    email: 'alejandro@lumapps.com'
  },
  method: 'GET',
  json: true,            
  headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${auth.access_token}`
  }, 
});
//Print the name of the connected user
console.log(JSON.stringify(response.fullName));
};


getToken();

File: Basics LumApps/python/oauth2.py
import httplib2
from googleapiclient.discovery import build
from oauth2client.client import OAuth2Credentials

CLIENT_ID = ''
CLIENT_SECRET = ''
REFRESH_TOKEN = ''


credentials = OAuth2Credentials(client_id=CLIENT_ID,
                                client_secret=CLIENT_SECRET,
                                refresh_token=REFRESH_TOKEN,
                                access_token=None,
                                token_expiry=None,
                                token_uri='https://accounts.google.com/o/oauth2/token',
                                user_agent=None)

http = httplib2.Http()
http = credentials.authorize(http)
accessTokenInfo = credentials.get_access_token(http)

service = build('lumsites', 'v1', http=http,
discoveryServiceUrl='https://lumsites.appspot.com/_ah/api/discovery/v1/apis/lumsites/v1/rest')

# Get the current user
currentUser = service.user().get().execute()
print ('Current user email: %s' % currentUser.get('email'))